### PR TITLE
crosscluster/logical: commit planner txn early

### PR DESCRIPTION
**Pull request by msbutler**
During large scale testing, we have seen the CreateForTables remote rpc take
several seconds or even minutes in LDR planning, and was previously within
scope of the planner txn, which could in turn cause txn refresh errors. This
patch commits the planner txn earlier in the job hook, before the
CreateForTables call, to prevent the planner txn from retrying. Now, the
destination side schema validation, locking, and job writing occur in a
subsequent txn.

Epic: none

Release note: none

### CODE REVIEW

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go [hunk 1]
```diff
@@ -178,6 +178,22 @@ func createLogicalReplicationStreamPlanHook(
 			}
 		}
 
+		// Commit the planner txn because several operations below may take several
+		// seconds, which we would like to conduct outside the scope of the planner
+		// txn to prevent txn refresh errors.
+		if err := p.Txn().Commit(ctx); err != nil {
```
Comment (dt) [inline pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:184]: did we already have a check above this that asserted the planner txn was implicit ?

**Comment by msbutler** [comment]
unrelated flake.

File: pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go (hunk 1 above)
Comment (msbutler) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:186]: oh lol, just found something.

Comment (msbutler) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:187]: fixed.

Comment (dt) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:187]: Wait, just to confirm, backup *does* check this, right?

Comment (msbutler) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:187]: on non detached backups:
https://github.com/msbutler/cockroach/blob/butler-different-default-values/pkg/ccl/backupccl/backup_planning.go#L559

Comment (dt) [reply pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go:187]: Ah, good good. Had me worried there for a minute. Carry on.

**Review by dt — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r=dt

**Pull request merged**